src/UI/Core/WindowStack.cpp

```diff
@@ -78,13 +75,18 @@ void WindowStack::updateStack(int x, int y)
 
 void WindowStack::bringToFront(Window* window)
 {
-	if (find(mWindowList.begin(), mWindowList.end(), window) == mWindowList.end())
+	const auto windowPosition = find(mWindowList.begin(), mWindowList.end(), window);
+	if (windowPosition == mWindowList.end())
 	{
 		std::cout << "WindowStack::bringToFront(): Window is not managed by this stack." << std::endl;
 		return;
 	}
+	if (windowPosition == mWindowList.begin())
+	{
+		return;
+	}
 
-	(*mWindowList.begin())->hasFocus(false);
+	mWindowList.front()->hasFocus(false);
 
 	mWindowList.remove(window);
 	mWindowList.push_front(window);
```

Comment:
Consider changing `mWindowList` to a `std::vector` instead. The number of windows on the stack is likely small enough that using a `std::list` is a waste due to cache locality misses.

Additionally, if using vector, consider swapping the target and front window.

If order matters, consider bubble sort to preserve the order by swapping items above to bubble up the target to the front. There is likely to be a small enough set size that the complexity of bubble sort won't matter.

Reply:
Funny, I was just thinking this sort of operation could be done more efficiently with a linked list rather than a vector. I wasn't aware it already was a linked list.

At any rate, this gets into the realm of micro optimizations. I wouldn't want to change it without some data to back it up.

Reply:
>...I wouldn't want to change it without some data to back it up.

https://isocpp.github.io/CppCoreGuidelines/CppCoreGuidelines#slcon2-prefer-using-stl-vector-by-default-unless-you-have-a-reason-to-use-a-different-container

:) Using `std::vector` instead of `std::list` for middle-insertions is explicitly mentioned for small (up to a few KB in size) containers. I highly doubt there will be a stack of 512 windows in use at once (assuming 4-byte integers and pointers being the same size; and the definition of "few" means more than 2.)

Like you said, by using a vector you can also increase efficiency by only calling `push_back` and reading the vector front-to-back with `.back()` being the "top" of the stack.

Removal is also much faster by using the swap-and-pop idiom. Swap the item to be removed with the last item in the list and call `pop_back`. Small container sizes can easily bubble the swapped item back to the top if order does matter.

Reply:
This is maybe getting a bit off topic for this PR.

I'm aware of that recommendation, and generally do use `std::vector` by default. Still, we have no actual performance data to say which would be better here. Asymptotically, a linked list is better, though as you've pointed out, we may never reach a large enough size for it to matter. But that's also a reason to not really care, since we'll never reach a large enough size for performance to matter.

Swap and pop is not suitable here, since you'd want to maintain the window ordering. Using swap and pop would cause the clicked on window to exchange places with the top most window, rather than rotate them in order.